ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+
+        if (retorn_printf > amplada)
+        {
+            int escurcament = retorn_printf - amplada;
+            if (escurcament > n_decimals)
+            {
+                *cadena = *MM_EmptyString;
+                return EOF;
+            }
+            *Error_sprintf_n_decimals = TRUE;
+            n_decimals = n_decimals - escurcament;
+            retorn_printf =
+                sprintf(cadena, "%*.*f", amplada, n_decimals, valor_double);
+        }
+        else
+            strcpy(cadena, cadena_treball);
+
+        return retorn_printf;
+    }
+
+    if (valor_double > VALOR_LIMIT_IMPRIMIR_EN_FORMAT_E ||
+        valor_double < -VALOR_LIMIT_IMPRIMIR_EN_FORMAT_E ||
+        (valor_double < VALOR_MASSA_PETIT_PER_IMPRIMIR_f &&
+         valor_double > -VALOR_MASSA_PETIT_PER_IMPRIMIR_f))
+    {
+        retorn_printf =
+            sprintf(cadena_treball, "%*.*E", amplada, n_decimals, valor_double);
+        if (retorn_printf == EOF)
+        {
+            *cadena = *MM_EmptyString;
+            return retorn_printf;
+        }
+        if (retorn_printf > amplada)
+        {
+            int escurcament = retorn_printf - amplada;
+            if (escurcament > n_decimals)
+            {
+                *cadena = *MM_EmptyString;
+                return EOF;
+            }
+            *Error_sprintf_n_decimals = TRUE;
+            n_decimals = n_decimals - escurcament;
+            retorn_printf =
+                sprintf(cadena, "%*.*E", amplada, n_decimals, valor_double);
+        }
+        else
+            strcpy(cadena, cadena_treball);
+
+        return retorn_printf;
+    }
+
+    retorn_printf =
+        sprintf(cadena_treball, "%*.*f", amplada, n_decimals, valor_double);
+    if (retorn_printf == EOF)
+    {
+        *cadena = *MM_EmptyString;
+        return retorn_printf;
+    }
+
+    if (retorn_printf > amplada)
+    {
+        int escurcament = retorn_printf - amplada;
+        if (escurcament > n_decimals)
+        {
+            *cadena = *MM_EmptyString;
+            return EOF;
+        }
+        *Error_sprintf_n_decimals = TRUE;
+        n_decimals = n_decimals - escurcament;
+        retorn_printf =
+            sprintf(cadena, "%*.*f", amplada, n_decimals, valor_double);
+    }
+    else
+        strcpy(cadena, cadena_treball);
+
+    return retorn_printf;
+
+#undef VALOR_LIMIT_IMPRIMIR_EN_FORMAT_E
+#undef VALOR_MASSA_PETIT_PER_IMPRIMIR_f
+}  // End of MM_SprintfDoubleWidth()
+
+static MM_BOOLEAN MM_EmptyString_function(const char *cadena)
+{
+    char *ptr;
+
+    for (ptr = (char *)cadena; *ptr; ptr++)
+        if (*ptr != ' ' && *ptr != '\t')
+            return FALSE;
+
+    return TRUE;
+}
+
+int MM_SecureCopyStringFieldValue(char **pszStringDst, const char *pszStringSrc,
+                                  MM_EXT_DBF_N_FIELDS *nStringCurrentLenght)
+{
+    if (!pszStringSrc)
+    {
+        if (1 >= *nStringCurrentLenght)
+        {
+            (*pszStringDst) = realloc_function(*pszStringDst, 2);
```

Comment:
can *pszStringDst be not-null before this function is called. If so, and if the realloc_function() fails, this will leak memory.
The safe pattern to use realloc and not leak memory is something like:
```
void* new_ptr = realloc(*pszStringDst, ...)
if( !new_ptr ) return ...;
*pszStringDst = new_ptr;
```

Reply:
Done